hotel/views.py

```diff
@@ -14,9 +18,122 @@ def user_register(request):
+                    availList.append(True)
+                else:
+                    availList.append(False)
+            if all(availList):
+                availableRooms.append(room)
+        return availableRooms
+
+    if request.method == 'POST':
+        if 'dateFilter' in request.POST:
+            check_in = request.POST.get("fd", "")
+            check_out = request.POST.get("ld", "")
+
+            check_in = datetime.strptime(check_in, '%Y-%m-%d')
+            check_out = datetime.strptime(check_out, '%Y-%m-%d')
+
+            rooms = check_valid(check_in, check_out)
+
+        if "filter" in request.POST:
+            if request.POST.get('id') != '':
+                rooms = rooms.filter(id=request.POST.get('id'))
+            if request.POST.get('capacity') != '':
+                rooms = rooms.filter(capacity__gte=request.POST.get('capacity'))
+            if request.POST.get('nob') != '':
+                rooms = rooms.filter(numberOfBeds__gte= request.POST.get('nob'))
+            if request.POST.get('type') != '':
+                rooms = rooms.filter(roomType__contains= request.POST.get('type'))
+            if request.POST.get('price') != '':
+                rooms = rooms.filter(room_price__lte= request.POST.get('price'))
+            context = {
+                "rooms": rooms,
+                "id": request.POST.get("id"),
+                "capacity": request.POST.get("capacity"),
+                "nob": request.POST.get("nob"),
+                "price": request.POST.get("price"),
+                "type": request.POST.get("type")
+            }
+            return render(request, 'room/rooms.html',context)
+    context = {
+        "rooms": rooms,
+        "fd": check_in,
+        "ld": check_out
+    }
+    return render(request, 'room/rooms.html', context)
+
+@login_required(login_url='login')
+def room_profile(request, number):
+    tempRoom = Room.objects.get(id=number)
+    img_urls = RoomImage.objects.filter(room_id= number)
+    bookings = Booking.objects.filter(room_id= tempRoom)
+    context = {
+        "bookings": bookings,
+        "room": tempRoom,
+        "img_urls": img_urls
+    }
+
+    return render(request,"room/room-profile.html", context)
+
+@login_required(login_url='login')
+def room_edit(request, number):
+    room = Room.objects.get(id= number)
+    form = editRoomForm(instance=room)
+
+    context = {
+        "room": room,
+        "form": form
+    }
+
+    if request.method == 'POST':
+        form = editRoomForm(request.POST, instance=room)
+        if form.is_valid():
+            form.save()
+            return redirect("room-profile", number = room.id)
+        else:
+            print(form.errors)
+    return render(request, "room/room-edit.html", context)
+
+@login_required(login_url='login')
+def room_add(request):
+    rooms_length = len(Room.objects.all())
+    if request.method == 'POST':
+        capacity = request.POST.get('capacity')
```

Comment:
ko có bước validate data, nên check data valid trước nhé
Add/Edit room em có thể dùng generic view nếu ko custom nhiều

Reply:
ở chỗ này, e check điều kiện nếu capacity < 1 or price <=0 or numOfBeds <1 và điều kiện capacity, numOfBeds không là số nguyên  thì sẽ in ra message là "Add room fail"
ở các file template e cũng để type của thẻ input đúng vs kiểu mình yêu cầu (chẳng hạn đặt type của capacity là number vs min = 1 , step = 1)